cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -4768,28 +4768,39 @@ mod tests {
                 "expected a variable that is valid in the policy scope; found: `foo`",
                 "must be one of `principal`, `action`, or `resource`"
```

Comment:
Let's update this error text match with the other error variant

Reply:
Updated the help text to match, but I think this error text describes the error well.  Do the changes look good to you?

Reply:
Seems fine. This would require a bit of work beyond the immediate scope of this PR, but the two error variants could probably be unified into one variant IMO